query/query.go

```diff
@@ -357,6 +357,12 @@ func (q *Branch) String() string {
 	return fmt.Sprintf("branch:%q", q.Pattern)
 }
 
+type RepoFunc func(name string) bool
+
+func (v RepoFunc) String() string {
+	return "RepoFunc"
+}
```

Comment:
should we take inspiration from MatchTree and include a reason to include as part of printing the matchtree/query